CI: Investigate failure of insync check `Orc8r Check Generated Files In Sync `

The workflow [`Orc8r Check Generated Files In Sync`](https://github.com/magma/magma/actions/workflows/insync-checkin.yml) (`insync-checkin.yml `) is frequently failing.

The [error messages](https://github.com/magma/magma/actions/runs/3949182013/jobs/6760070707) are:
```
make[1]: Leaving directory '/src/magma/orc8r/cloud'
# generate Swagger API bindings for NMS
/src/magma/nms/scripts/generateAPIFromSwagger.sh -f -i /src/magma/orc8r/cloud/go/services/obsidian/swagger/v1/swagger.yml -o /src/magma/nms/generated/
+ yarn --silent openapi-generator-cli version-manager set 6.0.0
warning package.json: No license field
Download 6.2.1 ...
Downloaded 6.2.1
Did set selected version to 6.2.1
Unable to query repository, because of: "Request failed with status code 504"

Response:
date Wed, 18 Jan 2023 13:18:44 GMT
content-type text/html
content-length 160
connection close
server nginx

/src/magma/orc8r/cloud/node_modules/@openapitools/openapi-generator-cli/main.js:1039
            error.response.data.on('data', data => this.logger.log(data.toString('utf8')));
                                ^

TypeError: error.response.data.on is not a function
    at VersionManagerService.printResponseError (/src/magma/orc8r/cloud/node_modules/@openapitools/openapi-generator-cli/main.js:1039:33)
    at /src/magma/orc8r/cloud/node_modules/@openapitools/openapi-generator-cli/main.js:918:18
    at /src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/operators/catchError.js:13:51
    at OperatorSubscriber._this._error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:43:21)
    at Subscriber.error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:60:18)
    at Subscriber._error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:84:30)
    at Subscriber.error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:60:18)
    at Subscriber._error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:84:30)
    at Subscriber.error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:60:18)
    at Subscriber._error (/src/magma/orc8r/cloud/node_modules/rxjs/dist/cjs/internal/Subscriber.js:84:30)

Node.js v18.13.0
make: *** [Makefile:191: nms_gen] Error 1
2
Creating build context in '/tmp/magma_orc8r_build'...
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/cloud/test/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/cloud/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/cloud/api/v1/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/gateway/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/lib/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/orc8r/lib/go/protos/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/lte/cloud/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/feg/cloud/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/feg/cloud/go/protos/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/cwf/cloud/go/go.mod
/tmp/magma_orc8r_build/gomod/src/magma/dp/cloud/go/go.mod
Running 'docker-compose build test'...
Running 'docker-compose run --rm -v /home/runner/work/magma/magma/.golangci.yml:/src/magma/.golangci.yml -v /home/runner/work/magma/magma/orc8r:/src/magma/orc8r -v /home/runner/work/magma/magma/lte:/src/magma/lte -v /home/runner/work/magma/magma/feg:/src/magma/feg -v /home/runner/work/magma/magma/cwf:/src/magma/cwf -v /home/runner/work/magma/magma/dp:/src/magma/dp test make fullgen'...
Error: Process completed with exit code 2.

```

--- a/nms/scripts/generateAPIFromSwagger.sh
+++ b/nms/scripts/generateAPIFromSwagger.sh
@@ -56,7 +56,15 @@ if [ "$FORCE" = false ] && [ -d "${OUTPUT}"  ] ; then
     fi
 fi
 
-(set -x; yarn --silent openapi-generator-cli version-manager set 6.0.0)
+if ! (set -x && yarn --silent openapi-generator-cli version-manager set 6.0.0) ;
+then
+  echo "Info: If this yarn command fails due to 'Request failed with status code 504'"
+  echo "the availability of search.maven.org may be responsible, see https://status.maven.org/#month"
+  echo "For more information see https://github.com/OpenAPITools/openapi-generator-cli/issues/680"
+  echo "or the issue https://github.com/magma/magma/issues/14860"
+  exit 1
+fi
+
 (set -x;
 yarn --silent openapi-generator-cli generate -i "${INPUT}" --output "${OUTPUT}" --skip-validate-spec --additional-properties=useSingleRequestParameter=true -g typescript-axios)
 

